### DISCUSSION TITLE
⚡ Surcharge is now available in Hyperswitch!

### DISCUSSION THREAD

**Discussion by hrithikesh026** [juspay/hyperswitch#2979] [Announcements]
### What is Surcharge?

- Surcharge is an additional fee applied during checkout, representing the payment processor's charge to the merchant per transaction.

### What can the feature do?

- The feature allows merchants to set surcharges during payment creation

- You can also establish rules for custom surcharges during payment confirmation based on parameters like amount, payment method, card network, etc.

Feel free to contribute and share your feedback. Here’s the [GitHub Link](https://github.com/juspay/hyperswitch)

**Comment by euluizaraujo** [comment]
@hrithikesh026 nice!  Is this already available in the open source version?  Is the configuration done in the open source control center?

**Reply by hrithikesh026** [reply]
Hi @euluizaraujo .
Apologies for the delayed response.
To answer your question, surcharge feature is available in open source version (can be tested through API). But configuration in control centre is actively under development.

**Reply by euluizaraujo** [reply]
@hrithikesh026 Could you give more details on how it works?  Is there any documentation?  I didn't find this functionality in the api references

**Reply by hrithikesh026** [reply]
Thank you for your interest @euluizaraujo . We are currently working on the documentation, will let you know once it is completed.

**Reply by euluizaraujo** [reply]
@hrithikesh026 perfect, thanks. So that we can explore the functionality, could you at least indicate where we find this function in the API?

**Reply by euluizaraujo** [reply]
@hrithikesh026

**Reply by bernard-eugine** [reply]
Hey Luiz. Expect this to be available by Dec 15.
